script/test-setup

```diff
@@ -5,13 +5,34 @@
 const fs = require('fs')
 const path = require('path')
 const cp = require('child_process')
-const { getUserDataPath } = require('./dist-info')
+const { getUserDataPath, getDistPath, getProductName } = require('./dist-info')
+
+const isFork = process.env.TRAVIS_SECURE_ENV_VARS !== 'true'
+
+if (process.platform === 'darwin' && process.env.TRAVIS && !isFork) {
+  const archive = `${getDistPath()}/${getProductName()}.app`
+  try {
+    console.log('validating signature of Desktop app')
+    cp.execSync(`codesign -dv --verbose=4 '${archive}'`)
+  } catch (err) {
+    process.exit(1)
+  }
+  console.log('\n\n')
+}
+
+if (process.platform === 'darwin' && process.env.TRAVIS) {
```

Comment:
The keychain prompt on launch occurs when it's hydrating the accounts it has in local storage - my gut feeling is some sort of state is floating around within the AppData folder to make it do this. Maybe some unit test isn't behaving itself?

I'll pull this out as we should still be good due to the codesigning fix and keep an eye on our tests.